ngraph/core/reference/include/ngraph/runtime/reference/detection_output.hpp

```diff
@@ -551,7 +551,7 @@ namespace ngraph
                     priorsBatchSize = priorsShape[0];
                     numLocClasses =
                         _attrs.share_location ? 1 : static_cast<size_t>(_attrs.num_classes);
-                    numResults = outShape[2];
+                    numResults = numImages * outShape[2];
```

Comment:
this should not be needed since num_images is already incorporated in outShape[2]: https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/op/detection_output.cpp#L254